hepcrawl/downloaders.py

```diff
@@ -0,0 +1,22 @@
+# -*- coding: utf-8 -*-
+#
+# This file is part of hepcrawl.
+# Copyright (C) 2016, 2017 CERN.
+#
+# hepcrawl is a free software; you can redistribute it and/or modify it
+# under the terms of the Revised BSD License; see LICENSE file for
+# more details.
+
+"""Additional downloaders."""
+
+
+from scrapy.http import Response
+
+
+class DummyDownloadHandler(object):
```

Comment:
this seems to be needed because for oai-pmh, you pass `oaipmh+http(s)` in the url. Cannot you just pass the real URL starting with `http(s)` instead and skip this whole download handler business?

Reply:
The issue is that we're using Sickle to do OAI stuff, and it does all of it's downloading on it's own, only exposing an iterator on the records. Since it doesn't expose any of it's requests/responses, there is not a nice way to integrate it with Scrapy so we bypass it.

Reply:
IMHO nope, because `scrapy` would otherwise try to fetch the URL with its default methods thus not allowing me to pass the ball to `sickle`.

Reply:
`XMLFeedSpider` seems to have no such problems...

Reply:
so IIUC, this whole contraption is done so that in the end, `start_requests` calls `parse`. If that's all you want, cannot you call `parse` from `start_requests` directly, given that you don't care about the response?

Reply:
I was trying something like this yesterday and was having issues. I didn't investigate very deeply, but I think Scrapy needs a request to begin crawling (either through `start_requests` or `url`). It expects `start_requests` to return a `Request` and doesn't allow anything.